backend/service_accounts/admin.py

```
@@ -1,15 +1,133 @@
+    )
+
+    def clean_scopes(self):
+        raw_scopes = re.split(r'[\s,]+', self.cleaned_data['scopes'].strip())
+        scopes = []
+        seen = set()
+        for scope in raw_scopes:
+            if scope and scope not in seen:
+                scopes.append(scope)
+                seen.add(scope)
+        if not scopes:
+            raise forms.ValidationError('Enter at least one scope.')
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)